src/ddmd/dscope.d

```diff
@@ -646,6 +647,30 @@ struct Scope
         return cast(Dsymbol)speller(ident.toChars(), &scope_search_fp, idchars);
     }
 
+    /************************************
+     * Maybe `ident` was a C or C++ name. Check for that,
+     * and suggest the D equivalent.
+     * Params:
+     *  ident = unknown identifier
+     * Returns:
+     *  D identifier string if found, null if not
+     */
+    extern (C++) static const(char)* search_correct_C(Identifier ident)
```

Comment:
I used the same convention as the existing `search_correct` to tie the two together.